packages/vscode-extension/src/handlers.ts

```diff
@@ -584,10 +585,18 @@ export async function addCapabilityHandler(args: any[]): Promise<Result<null, Fx
     VsCodeLogInstance.warning(`${error}`);
   }
   const result = await runUserTask(func, TelemetryEvent.AddCap, true);
-  if (result.isOk()) {
-    await globalStateUpdate("automaticNpmInstall", true);
-    automaticNpmInstallHandler(excludeFrontend, true, excludeBot);
+  if (result.isErr()) {
+    return err(result.error);
   }
+
+  // refresh env tree after adding capability.
+  const refreshResult = await registerEnvTreeHandler();
+  if (refreshResult.isErr()) {
+    return err(refreshResult.error);
+  }
+
+  await globalStateUpdate("automaticNpmInstall", true);
```

Comment:
If addCapability succeed but refresh tree view failed, the global state will be affected. What about move it above envTree refresh?

Reply:
Updated~